### PR TITLE
Add reports from pytest-mpl, changed setup and tox

### LINKED ISSUE #97: pytest-mpl report out from tox and upload it as a build artefact in the CI

### PR DESCRIPTION
**Pull request by ruaridhg**
Closes #97 

- 2D and 3D pytest-mpl reports added for histogram and slice with html summary `fig_comparison.html`
- matplotlib added to setup.cfg testing
- cp and ls commands added to tox and mpl-result-path set in tox

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -54,6 +54,7 @@ docs =
     sphinx-automodapi
     sphinx-gallery
 testing =
+    matplotlib
```
Comment (dstansby) [inline setup.cfg:57]: This doesn't need to be here, since it's already an installation dependency.

File: src/napari_matplotlib/reports/extra.js
Comment (dstansby) [inline src/napari_matplotlib/reports/extra.js:1]: I don't think we want to commit this file?

**Review by dstansby — CHANGES_REQUESTED** [review]
We don't want to commit an individual report from a test run to the repository - the report should be generated during testing (either locally or on GitHub actions), and then we want to upload that as an artefact at the end of the GitHub actions run.

So I think the TODO here is (let me know if this doesn't make sense or you think otherwise!):
- Undo adding the `report` directory, and put back `report` in gitignore
- Add a step at the end of the GH actions run that uploads the whole `report` directory using [actions/upload-artifact](https://github.com/actions/upload-artifact)

**Comment by ruaridhg** [comment]
> We don't want to commit an individual report from a test run to the repository - the report should be generated during testing (either locally or on GitHub actions), and then we want to upload that as an artefact at the end of the GitHub actions run.
> 
> So I think the TODO here is (let me know if this doesn't make sense or you think otherwise!):
> 
> * Undo adding the `report` directory, and put back `report` in gitignore
> * Add a step at the end of the GH actions run that uploads the whole `report` directory using [actions/upload-artifact](https://github.com/actions/upload-artifact)

I was actually just thinking the same thing so will add the upload-artifact step

**Comment by ruaridhg** [comment]
Closes #97

**Review by dstansby — APPROVED** [review]
Noice, looking good!
Screenshot 2023-05-19 at 16 54 28

**Pull request merged**